lib/sync.js

```diff
@@ -655,6 +655,8 @@ SyncApi.prototype._processSyncResponse = function(syncToken, data) {
         }
     }
 
+    this.notifEvents = [];
```

Comment:
ugh in terms of abusing a member field, mostly. but the other changes help ameliorate this.